src/platforms/hosted/jlink.c

```diff
@@ -42,48 +43,71 @@
 
 #include "cli.h"
 
-#define USB_VID_SEGGER 0x1366U
+typedef struct jlink {
+	char fw_version[256U];         /* Firmware version string */
+	uint32_t hw_version;           /* Hardware version */
+	uint32_t capabilities;         /* Bitfield of supported capabilities */
+	uint32_t available_interfaces; /* Bitfield of available interfaces */
 
-#define USB_PID_SEGGER_0101 0x0101U
-#define USB_PID_SEGGER_0105 0x0105U
-#define USB_PID_SEGGER_1015 0x1015U
-#define USB_PID_SEGGER_1020 0x1020U
+	struct jlink_interface_frequency {
+		uint32_t base;            /* Base frequency of the interface */
+		uint16_t min_divisor;     /* Minimum divisor for the interface */
+		uint16_t current_divisor; /* Current divisor for the interface */
+	} interface_frequency[JLINK_INTERFACE_MAX];
+} jlink_s;
 
-static uint32_t jlink_caps;
-static uint32_t jlink_freq_khz;
-static uint16_t jlink_min_divisor;
-static uint16_t jlink_current_divisor;
-uint8_t jlink_interfaces;
+jlink_s jlink;
 
-int jlink_simple_query(const uint8_t command, void *const rx_buffer, const size_t rx_len)
+/* J-Link USB protocol functions */
+
+bool jlink_simple_query(const uint8_t command, void *const rx_buffer, const size_t rx_len)
+{
+	return bmda_usb_transfer(info.usb_link, &command, sizeof(command), rx_buffer, rx_len, JLINK_USB_TIMEOUT) >= 0;
+}
+
+bool jlink_simple_request(const uint8_t command, const uint8_t operation, void *const rx_buffer, const size_t rx_len)
```

Comment:
Given the below two functions, it would make sense for consistency for this to be `jlink_simple_request_8` we would suggest.